### ISSUE TITLE
[BUG] [PYTHON CLIENT] response is None although data is returned HTTP 2xx

### ISSUE THREAD

**Issue by IAmWebSA** [OpenAPITools/openapi-generator#12189]
Running openapi cli generator for python results in 'response_type': None for POST although multiple are defined.

The Route can return the following status codes 200, 202, 401, 403 or 500.
The issue is if the route is called via the python client it returns "None" for the 202 status code instead of the actual returned value of "{id:123}". 

Is there away to let the client handle also such 2xx responses to return the data?

Via other REST Tools it displays the correct response values as application/json as it should be.

Sample definitions:

****
```
openapi: 3.0.1
info:
  title: REST API
  version: "0.1"
paths:
  /test:
    post:
      summary: Trigger a async task
      description: Trigger a async task
      responses:
        200:
          description: OK
          content: {}
        202:
          description: route executed and ID returend.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Response202'
        401:
          description: ohno
          content: {}
        403:
          description: ohno
          content: {}
        500:
          description: Error
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Error'


components:
  schemas:
    Error:
      type: object
      properties:
        help:
          type: string
    Response202:
      required:
      - Id
      type: object
      properties:
        Id:
          type: integer
          description: The id
          example: 1
```

****

**Comment by spacether** [comment]
Can you please try python-experimental? It has better deserialization support than python.

**Comment by IAmWebSA** [comment]
1.
I have tried with 5.4,0 and 6.0.0 Beta using:

java -jar openapi-generator-cli.jar generate -i "openapi3.yaml" -g python-experimental -o openapi-python-client --package-name rest_client --additional-properties packageVersion=0.1,packageName=rest_client,modelPropertyNaming=original


But now it is even worse and has syntax issues:



**Error : "Union requires two or more type argumentsPylance"**
```
@dataclass
class ApiResponseFor200(api_client.ApiResponse):
    response: urllib3.HTTPResponse
    body: typing.Union[
    ]
    headers: Unset = unset

```
2.
What I also mentioned is that if you have a HTTP Response 200 with a required field set and you get a HTTP Response 202 with a different return structure it throws an error that it cannot find the required property from the 200 response in the 202 response.

Is this also a known bug?

3.
I also tried to get rid og the snake_case conversion with the "modelPropertyNaming=original" but this also does not work.
Am I doing something wrong?
If I have a property like MyCamelProperty it converts the returned property to "my_camel_property", this is strange to me and has to be prevented.
